tracing-subscriber/src/fmt/format/pretty.rs

```diff
@@ -426,6 +426,17 @@ impl<'a> PrettyVisitor<'a> {
 }
 
 impl<'a> field::Visit for PrettyVisitor<'a> {
+    fn record_byte_slice(&mut self, field: &Field, value: &[u8]) {
+        let hex =
+            value
+                .iter()
+                .fold(String::with_capacity(value.len() * 2), |mut acc, byte| {
```

Comment:
it's kind of unfortunate that this has to allocate a string. i think we could write a wrapper type for a byte slice with a `Debug` impl that records the bytes in hex, and pass that to `record_debug`, rather than allocating?

Reply:
True. The default implementation does not allocate and since it did the same thing  as this one, I deleted this override. I also deleted the json one since the default falls to debug so it gets printed correctly as a string with the hex representation.